cmake/kokkos_arch.cmake

```diff
@@ -845,15 +845,42 @@ IF (KOKKOS_ENABLE_OPENACC)
       Clang -Xopenmp-target=nvptx64-nvidia-cuda -march=${CLANG_CUDA_ARCH}
             -fopenmp-targets=nvptx64-nvidia-cuda
     )
+    IF(DEFINED ENV{CUDA_PATH})
+      COMPILER_SPECIFIC_LINK_OPTIONS(Clang -L$ENV{CUDA_PATH}/lib64)
+    ENDIF()
+    COMPILER_SPECIFIC_LIBS(
+      Clang -lcudart
+      NVHPC -cuda
+    )
   ELSEIF(KOKKOS_AMDGPU_ARCH_FLAG)
     COMPILER_SPECIFIC_FLAGS(
       Clang -Xopenmp-target=amdgcn-amd-amdhsa -march=${KOKKOS_AMDGPU_ARCH_FLAG}
             -fopenmp-targets=amdgcn-amd-amdhsa
     )
+    IF(DEFINED ENV{ROCM_PATH})
+      COMPILER_SPECIFIC_FLAGS(Clang -I$ENV{ROCM_PATH}/include)
+      COMPILER_SPECIFIC_LINK_OPTIONS(Clang -L$ENV{ROCM_PATH}/lib)
+    ENDIF()
+    COMPILER_SPECIFIC_LIBS(Clang -lamdhip64)
+  ELSEIF(KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE)
+    # Compile for kernel execution on the host. In that case,
+    # memory is shared between the OpenACC space and the host space.
+    COMPILER_SPECIFIC_FLAGS(
+      NVHPC -acc=multicore
+    )
+    COMPILER_SPECIFIC_DEFS(
+      NVHPC KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE
+    )
+    COMPILER_SPECIFIC_DEFS(
+      Clang KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE
+    )
```

Comment:
Remove and add them to `cmake/KokkosCore_config.h.in` instead

Reply:
`KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE` is supposed to be enabled only when none of NVIDIA and AMD GPU architecture is specified. If we add it to `cmake/kokkosCore_config.h.in`, it will be defined regardless of GPU architecture setting.

Reply:
Then you could/should have that cmakedefine in the config file and error out at configuration time if both an NVIDIA/AMD GPU arch and that FORCE_HOST_AS_DEVICE option are enabled

Reply:
Updated CMake/Make so that it will be added to `KokkosCore_config.h` when specified. 
CMake will print a warning message if both `KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE` and GPU architecture are specified, and the OpenACC backend will ignore  `KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE` in that case.